jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/PlatformSpec.java

```diff
@@ -85,15 +86,15 @@ public PlatformSpec(
     return Optional.ofNullable(osVersion);
   }
 
-  public Optional<List<String>> getOsFeatures() {
-    return Optional.ofNullable(osFeatures);
+  public List<String> getOsFeatures() {
```

Comment:
Sorry, can we return ImmutableList<String> here instead of List? Might be more secure to pass around immutable collections.

Reply:
That's a good question. Usually we use the base type interface on return values. That keeps implementation details from leaking out. However having said that, this is not an api that is exposed externally, so it could be okay. The other issue is that we don't know what kind of list the yaml parser returns, so we would have to wrap the list generated in the constructor  (https://github.com/GoogleContainerTools/jib/pull/2609/files#diff-784a41ae78bc030a7b61cd50c6a28e42R72) with a ImmutableList type.

So there's a lot going on here, and there's no obvious correct answer. It could go either way. In general though, we use the base type interface, and I just happened to continue to use that pattern.

Reply:
Thanks for the explanation! That makes sense:)